app/src/main/java/ch/epfl/polybazaar/UI/SaleDetails.java

```diff
@@ -187,33 +164,40 @@ private void retrieveImages(String listingID) {
     private void drawImages() {
         runOnUiThread (()-> {
             List<SliderItem> sliderItems = new ArrayList<>();
-            for(String strImg: listStringImage) {
-                sliderItems.add(new SliderItem(convertStringToBitmap(strImg)));
-            }
+            if (!listStringImage.isEmpty()) {
+                viewPager2.setVisibility(View.VISIBLE);
+                imageLoading.setVisibility(View.GONE);
+                findViewById(R.id.pageNumber).setVisibility(View.VISIBLE);
+                for (String strImg : listStringImage) {
+                    sliderItems.add(new SliderItem(convertStringToBitmap(strImg)));
+                }
 
-            viewPager2.setAdapter(new SliderAdapter(sliderItems, viewPager2));
+                viewPager2.setAdapter(new SliderAdapter(sliderItems, viewPager2));
 
-            viewPager2.setClipToPadding(false);
-            viewPager2.setClipChildren(false);
-            viewPager2.setOffscreenPageLimit(3);
-            viewPager2.getChildAt(0).setOverScrollMode(RecyclerView.OVER_SCROLL_NEVER);
+                viewPager2.setClipToPadding(false);
+                viewPager2.setClipChildren(false);
+                viewPager2.setOffscreenPageLimit(3);
+                viewPager2.getChildAt(0).setOverScrollMode(RecyclerView.OVER_SCROLL_NEVER);
 
-            CompositePageTransformer compositePageTransformer = new CompositePageTransformer();
-            compositePageTransformer.addTransformer((page, position) -> {
-                float r = 1 - Math.abs(position);
-                page.setScaleY(0.85f + r * 0.15f);
-            });
-            viewPager2.setPageTransformer(compositePageTransformer);
-
-            viewPager2.registerOnPageChangeCallback(new ViewPager2.OnPageChangeCallback() {
-                @Override
-                public void onPageSelected(int position) {
-                    super.onPageSelected(position);
-                    TextView textPageNumber = findViewById(R.id.pageNumber);
-                    textPageNumber.setText(String.format("%s/%s", Integer.toString(viewPager2.getCurrentItem() + 1), Integer.toString(listStringImage.size())));
-                    textPageNumber.setGravity(Gravity.CENTER);
-                }
-            });
+                CompositePageTransformer compositePageTransformer = new CompositePageTransformer();
+                compositePageTransformer.addTransformer((page, position) -> {
+                    float r = 1 - Math.abs(position);
+                    page.setScaleY(0.85f + r * 0.15f);
```

Comment:
Would it be possible to have some clarification about this formula ? What do these parameters do ?

Reply:
I didn't write that, so I really can't say.